react-backend/client/src/components/admin/adminVisitsPage.js

```diff
@@ -92,19 +150,112 @@ export class AdminVisitsPage extends Component {
 
   passwordError = <span>The password is incorrect. Please try again.</span>;
 
-  render() {
-    return this.state.reauthenticated ? (
+  renderAuthenticated = () => {
+    return (
       <div>
         <h1>Visitor Data</h1>
-        <select onChange={this.filter}>
-          <option defaultValue value="date">
+        <form>
+          <label className="Form__Label">
             Sort by
-          </option>
-          <option value="yearofbirth">Year of Birth </option>
-          <option value="sex">Gender </option>
-          <option value="activity">Activity </option>
-          <option value="date">Date of visit </option>
-        </select>
+            <br />
+            <select onChange={this.sort}>
+              <option defaultValue value="date">
+                Sort by
+              </option>
+              <option value="yearofbirth">Year of Birth </option>
+              <option value="sex">Gender </option>
+              <option value="activity">Activity </option>
+              <option value="date">Date of visit </option>{' '}
+            </select>
+          </label>
+          <label className="Form__Label">
+            Filter by Gender
+            <br />
+            <label>
+              <input
+                type="checkbox"
+                value="male"
+                onChange={this.filter.bind(this, 'gender')}
```

Comment:
All of these `bind`s can be changed after the above comment about partial application